### PR TITLE
HBASE-23686 Revert binary incompatible change in ByteRangeUtils and removed reflections in CommonFSUtils

### PR DESCRIPTION
**Pull request by HorizonNet**
Based on the feedback from Sean on #1034.

### CODE REVIEW

File: hbase-common/src/main/java/org/apache/hadoop/hbase/util/CommonFSUtils.java [hunk 1]
```diff
@@ -568,83 +521,54 @@ static void setStoragePolicy(final FileSystem fs, final Path path, final String
...
-                  "setStoragePolicy. This is probably intentional on their part, since HDFS-9345 " +
-                  "appears to be present in your version of Hadoop. For more information check " +
-                  "the Hadoop documentation on 'ArchivalStorage', the Hadoop FileSystem " +
-                  "specification docs from HADOOP-11981, and/or related documentation from the " +
-                  "provider of the underlying FileSystem (its name should appear in the " +
-                  "stacktrace that accompanies this message). Note in particular that Hadoop's " +
-                  "local filesystem implementation doesn't support storage policies.", exception);
-            }
+      // check for lack of HDFS-7228
+      if (e instanceof InvocationTargetException) {
```
Comment (busbey) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/util/CommonFSUtils.java]: This should be two direct checks for `RemoteException` and `UnsupportedOperationException` respectively. Now that we're not using reflection it won't be wrapped in `InvocationTargetException`

Comment (HorizonNet) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/util/CommonFSUtils.java]: Removed the outer check for the `InvocationTargetException`.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 39s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 20s |  master passed  |
| +1 :green_heart: |  compile  |   0m 39s |  master passed  |
| +1 :green_heart: |  checkstyle  |   3m  2s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  3s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 51s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 16s |  branch/hbase-checkstyle no findbugs output file (findbugsXml.xml)  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   4m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 43s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed  |
| +0 :ok: |  findbugs  |   0m 13s |  hbase-checkstyle has no data from findbugs  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 14s |  hbase-checkstyle in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 20s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 164m 39s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  0s |  The patch does not generate ASF License warnings.  |
|  |   | 227m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1036/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1036 |
| Optional Tests | dupname asflicense checkstyle javac javadoc unit xml spotbugs findbugs shadedjars hadoopcheck hbaseanti compile |
| uname | Linux 1094328c8842 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1036/out/precommit/personality/provided.sh |
| git revision | master / 499ff32f00 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1036/1/testReport/ |
| Max. process+thread count | 5729 (vs. ulimit of 10000) |
| modules | C: hbase-checkstyle hbase-common U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1036/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 27s |  master passed  |
| +1 :green_heart: |  compile  |   0m 38s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m 38s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 41s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 48s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 16s |  branch/hbase-checkstyle no findbugs output file (findbugsXml.xml)  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 36s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
| +0 :ok: |  findbugs  |   0m 14s |  hbase-checkstyle has no data from findbugs  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 14s |  hbase-checkstyle in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 19s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 153m 29s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  0s |  The patch does not generate ASF License warnings.  |
|  |   | 211m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1036/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1036 |
| Optional Tests | dupname asflicense checkstyle javac javadoc unit xml spotbugs findbugs shadedjars hadoopcheck hbaseanti compile |
| uname | Linux bbfc7495a56d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1036/out/precommit/personality/provided.sh |
| git revision | master / 499ff32f00 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1036/2/testReport/ |
| Max. process+thread count | 5535 (vs. ulimit of 10000) |
| modules | C: hbase-checkstyle hbase-common U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1036/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by HorizonNet** [comment]
Problem in the build seems to be related to [HBASE-22732](https://issues.apache.org/jira/browse/HBASE-22732]).

**Pull request merged**